Add companion whitepaper: Operator Token Acquisition for Number Verification on Android (TS.43/OpenID4VP) (#238)
camaraproject/NumberVerification: documentation/SupportingDocuments/CAMARA Number Verification — Operator Token Acquisition.md

**Request (Application → Aggregator)**

```http
POST /token HTTP/1.1
Host: <aggregator-host>
Authorization: <aggregator-specific>
Content-Type: application/x-www-form-urlencoded

grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Ajwt-bearer
&assertion=<JWT signed by Application>
```

> **Parameter placement:** `grant_type` and `assertion` travel in the **form-urlencoded request body** (per RFC 6749 §3.2 for OAuth `/token` endpoints). `Authorization` and `Content-Type` are HTTP headers. Bearer credentials SHALL NOT appear in the query string or path.

| Field | Description |
|---|---|
| `grant_type` | Fixed: `urn:ietf:params:oauth:grant-type:jwt-bearer` (RFC 7523 — names the bearer-credential grant family). |
| `assertion` | JWT Bearer assertion (RFC 7523) built and signed by the Application with its private key registered with the Aggregator. Required claims: `iss` = Application's `client_id` at the Aggregator; `sub` = `"vp_token:<SD-JWT>"` (the full SD-JWT from Step 2, with prefix); `aud` = this Aggregator's `/token` endpoint URL; `scope` = CAMARA scope (MUST be present as a JWT claim — MUST NOT appear as a separate form body parameter); `exp`, `iat`, `jti` per RFC 7523. |

> > **`sub` claim — Operator Token as grant subject:** The Operator Token (SD-JWT from Step 2) is placed in the `sub` claim of the assertion using the prefix `"vp_token:<SD-JWT>"` (renamed from `operatortoken:` following the PR #238 review, reflecting that the value carried is the OpenID4VP `vp_token`). The `scope` claim MUST be inside the assertion JWT; it MUST NOT appear in the request body. The Aggregator verifies the Application's signature, extracts the SD-JWT from `sub`, and after JWE decryption builds its own assertion to the CSP with the prefix `"operatortoken:<temp_token>"` (§5.3).

Author: ```suggestion
> **`sub` claim — Operator Token as grant subject:** The Operator Token (SD-JWT from Step 2) is placed in the `sub` claim of the assertion using the prefix `"vp_token:<SD-JWT>"`. The `scope` claim MUST be inside the assertion JWT; it MUST NOT appear in the request body. The Aggregator verifies the Application's signature, extracts the SD-JWT from `sub`, and after JWE decryption builds its own assertion to the CSP with the prefix `"operatortoken:<temp_token>"` (§5.3).
```

```json
{ "devicePhoneNumber": "+34XXXXXXXXX" }
```

**Verify phone number**

```http
POST /number-verification/v2/verify HTTP/1.1
Host: <aggregator-host>
Authorization: Bearer <access_token>
Content-Type: application/json

{ "phoneNumber": "+34XXXXXXXXX" }
```

> **Parameter placement:** `phoneNumber` travels in the **JSON request body**. The access token is in the `Authorization` HTTP header. Nothing in the query string.

```json
{ "devicePhoneNumberVerified": true }
```

---

## 5. Aggregator (Channel Partner) Internal Responsibilities

This section describes what the Aggregator SHALL do internally upon receiving the JWT Bearer assertion from the Application in Step 3. From the Application's perspective Step 3 is a single round-trip (`/token` in, access token out); everything below — SD-JWT validation, JWE decryption, the RFC 7523 call to the CSP — happens inside the Aggregator before it responds to the App. The Aggregator → CSP interface (§5.3) **is** described by this guide as a recommended, CSP-agnostic flow; the Application developer does not implement any of this logic.

### 5.1 Application assertion validation and SD-JWT processing

> The JWT Bearer grant used at this interface is defined in [CAMARA Identity and Consent Management — JWT Bearer Flow](https://github.com/camaraproject/IdentityAndConsentManagement/blob/r4.2/documentation/CAMARA-API-access-and-user-consent.md#jwt-bearer-flow). This section specifies the additional validation steps required for the `vp_token:` subject prefix and the SD-JWT structure carried in the `sub` claim:

Author: ```suggestion
The App → Aggregator interface described in this section is **Aggregator-specific and is not defined by CAMARA** — GSMA Open Gateway allows the Aggregator to expose a CAMARA-shaped interface or its own. It uses a JWT Bearer assertion (RFC 7523) with the `vp_token:` subject prefix defined by this guide. The CAMARA [JWT Bearer Flow with Operator Token](https://github.com/camaraproject/IdentityAndConsentManagement/blob/r4.2/documentation/CAMARA-API-access-and-user-consent.md#jwt-bearer-flow-with-operator-token) applies to the Aggregator → CSP hop (§5.3), not to this interface. This section specifies the validation steps the Aggregator performs on the `vp_token:` subject prefix and the SD-JWT structure carried in the `sub` claim:
```

**Nonce and anti-replay.** The nonce is generated by the Application in Step 1, embedded in the `credential_authorization_jwt` by the Aggregator, and carried inside the Binding JWT (signed by the device key). The Aggregator SHALL validate the nonce in Step 3 and SHALL reject any SD-JWT whose nonce has already been consumed. Each Operator Token is therefore single-use.

**SD-JWT integrity.** The Issuer JWT is signed by Android Telephony using a carrier-issued key conveyed in the `x5c` header chain. The Binding JWT is signed by an ephemeral device key whose public key appears in the `cnf` claim of the Issuer JWT. The Aggregator SHALL verify both signatures (§5.1) before any further processing.

**Consent auditability.** The `consent_data_hash` field in the Binding JWT is the SHA-256 of the consent material shown to the user. Because it is covered by the device's signature, it constitutes a cryptographic attestation that the correct consent text was presented before the token was issued.

> 

Reviewer: ```suggestion

**Aggregator auditing and revocation.** CSPs SHALL establish ongoing security-audit obligations for contracted Aggregators as a condition of the bilateral agreement and GMS whitelist entry. When a compromise is confirmed, two independently sufficient revocation mechanisms are available:

- **GMS whitelist revocation** — the CSP instructs Android to remove the Aggregator's certificate fingerprint from the GMS whitelist (Appendix B.1). GMS will then refuse to issue Operator Tokens for that Aggregator on any device.
- **CAMARA Auth Server revocation** — the CSP revokes the Aggregator's `client_id` registration at the CAMARA Auth Server. Subsequent JWT Bearer assertions from that Aggregator are rejected at Step 3 (§5.3) regardless of GMS whitelist state.

Either mechanism is independently sufficient to block a compromised Aggregator; operating both provides defence in depth.

**Play Integrity API (defence in depth).** Aggregators SHOULD verify the integrity of the requesting device and application environment using the [Google Play Integrity API](https://developer.android.com/google/play/integrity) or an equivalent hardware-backed device-attestation mechanism. The verdict MAY be evaluated at Step 1 (before issuing the `credential_authorization_jwt`) or at Step 3 (before accepting the SD-JWT). A failed device-integrity verdict SHOULD result in the credential being denied or the request being flagged for elevated risk review.

```

- It verifies the JWS signature with the public key of the leaf certificate.
- Only if both checks pass does GMS proceed to invoke EAP-AKA against the Entitlement Configuration Server (ECS) and produce a `temp_token`.

This mechanism is what allows each CSP to **gate, per-Aggregator, who may obtain Operator Tokens for its subscribers**. It is also why an Aggregator SHALL be onboarded with every CSP whose subscribers it intends to serve — there is no general "any-CSP" credential. The Aggregator's signing private key is therefore a high-value secret and SHOULD be held in an HSM or equivalent.

> **What the Application is *not*:** the Application has no certificate, no key, and no whitelist entry on the device side. From the Wallet's perspective only the Aggregator is cryptographically authorised. The Application's identity reaches the CSP only later, in §5.3, via the `iss = aggregator_client_id_at_csp` claim of the JWT Bearer assertion — where `aggregator_client_id_at_csp` is the client ID the Application (or its Aggregator acting on its behalf) has registered with that CSP during onboarding.

> 

Reviewer: ```suggestion

**Ongoing governance obligations.** Onboarding establishes the initial trust anchor; maintaining ecosystem security requires CSPs to treat Aggregator management as a continuous operational responsibility:

- CSPs SHALL conduct periodic security auditing of all contracted Aggregators — including key-management practices, token-handling procedures, access logging, and incident-response readiness.
- CSPs SHALL maintain operational readiness to invoke GMS whitelist revocation (via Android) and CAMARA Auth Server revocation independently of each other, so that either layer can block a compromised Aggregator without dependency on the other.
- The bilateral Aggregator agreement SHALL include contractual provisions for immediate whitelist suspension upon confirmed malicious activity or material security-policy violation.

```
